app/src/org/commcare/android/models/AndroidSessionWrapper.java

```diff
@@ -131,72 +125,6 @@ public void setFormRecordId(int formRecordId) {
         this.formRecordId = formRecordId;
     }
     
-    /**
-     * Registers the instance data returned from form entry about this session, and specifies
-     * whether the returned data is complete 
-     *
-     * @param uri points to the instance we want to register with the session
-     * @param c A cursor which points to at least one record of an ODK instance.
-     * @return True if the record in question was marked completed, false otherwise
-     * @throws IllegalArgumentException If the cursor provided doesn't point to any records,
-     * or doesn't point to the appropriate columns
-     */
-    public boolean beginRecordTransaction(Uri uri, Cursor c) throws IllegalArgumentException {
-        if (!c.moveToFirst()) {
-            throw new IllegalArgumentException("Empty query for instance record!");
-        }
-
-        // set local state
-        instanceUri = uri.toString();
-        instanceStatus = c.getString(c.getColumnIndexOrThrow(InstanceColumns.STATUS));
-
-        // was the record marked complete?
-        return InstanceProviderAPI.STATUS_COMPLETE.equals(instanceStatus);
-    }
-
-    /**
-     * Update the session's form record status and link the record to an instance.
-     */
-    public FormRecord commitRecordTransaction() throws InvalidStateException {
```

Comment:
Moved this to InstanceProvider and renamed to syncRecordToInstance, merging logic from beginRecordTransaction